### PR TITLE
Use image's digest instead of binary's checksum 

### PR DESCRIPTION
**Pull request by johnlanni**
Fixes: #37565

### CODE REVIEW

**Comment by johnlanni** [comment]
@bianpengyuan  Implemented as discussed [here](https://github.com/istio/istio/issues/37565).

**Review by zirain — COMMENTED** [review]
LGTM, leave to @bianpengyuan

File: pkg/wasm/cache.go [hunk 1]
```diff
@@ -157,7 +157,7 @@ func (c *LocalFileCache) Get(downloadURL, checksum string, timeout time.Duration
 		}
 		wasmLog.Debugf("wasm oci fetch %s with options: %v", downloadURL, imgFetcherOps)
 		fetcher := NewImageFetcher(ctx, imgFetcherOps)
-		b, err = fetcher.Fetch(u.Host+u.Path, checksum)
+		b, dChecksum, err = fetcher.Fetch(u.Host+u.Path, checksum)
```
Comment (bianpengyuan) [inline pkg/wasm/cache.go:160]: If this function does not return error, the checksum should be matched right? So we don't really need to return the checksum, just `dChecksum = checksum` would be good?

Comment (johnlanni) [reply pkg/wasm/cache.go:160]: Need to consider the case where the `sha256` field of the WasmPlugin resource is not set, and the checksum parameter here is an empty string.

Comment (bianpengyuan) [reply pkg/wasm/cache.go:160]: Ah I see thanks!

**Review by bianpengyuan — APPROVED** [review]

**Pull request merged**